src/App.tsx

```diff
@@ -1,93 +1,117 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useContext } from 'react';
+import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Error } from './Error';
+import { TodoList } from './TodoList';
+import { Header } from './Header';
+import { Footer } from './Footer';
+import { TodosContext } from './TodoContext';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
 
-export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+type Props = {
+  visibleTodos: Todo[];
+  filter: FilterStatus,
+  setFilter(filterType: FilterStatus): void,
+};
 
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
+const USER_ID = 9968;
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+export const App: React.FC<Props> = ({
+  visibleTodos,
+  filter,
+  setFilter,
+}) => {
+  const {
+    todos,
+    setTodos,
+    isDeleteError,
+    setIsDeleteError,
+    isAddError,
+    setIsAddError,
+    isUpdateError,
+    setIsUpdateError,
```

Comment:
Why you need so mush error states? 
Use one that will have error message that you will show to user or null in case when no error